mithril-client-cli/src/commands/cardano_db/download.rs

```diff
@@ -60,23 +60,52 @@ pub struct CardanoDbDownloadCommand {
 }
 
 impl CardanoDbDownloadCommand {
-    /// Is JSON output enabled
-    pub fn is_json_output_enabled(&self) -> bool {
-        self.shared_args.json
-    }
-
     /// Command execution
     pub async fn execute(&self, context: CommandContext) -> MithrilResult<()> {
+        let params: ConfigParameters = context.config_parameters()?.add_source(self)?;
+        let prepared_command: PreparedCardanoDbDownload = self.prepare(&params)?;
+
+        prepared_command.execute(context.logger(), params).await
+    }
+
+    /// Verifies the command's configuration and prepare a strucutred object for execution
+    fn prepare(&self, params: &ConfigParameters) -> MithrilResult<PreparedCardanoDbDownload> {
```

Comment:
This would avoid to clone all the fields to create the new structure:
```suggestion
    fn prepare(self, params: &ConfigParameters) -> MithrilResult<PreparedCardanoDbDownload> {
```

Reply:
@jpraynaud Yes but as the `self` is also a shared reference in `execute` function of `CardanoDbDownloadCommand`, I will have to clone self before calling `prepare` , is it acceptable ? 
```
impl CardanoDbDownloadCommand {
    /// Command execution
    pub async fn execute(&self, context: CommandContext) -> MithrilResult<()> {
        let params: ConfigParameters = context.config_parameters()?.add_source(self)?;
        let prepared_command: PreparedCardanoDbDownload = self.clone().prepare(&params)?;

        prepared_command.execute(context.logger(), params).await
    }
...
```

Reply:
You can probably try to consume the command in the execute function: `pub async fn execute(self, context: CommandContext) -> MithrilResult<()>` instead.

Reply:
as discussed together, we keep the .clone() at attribute level for now